### PR TITLE
Upgrade to newer Sass API.

### LINKED ISSUE #218: Support newer Sass JS API.
https://sass-lang.com/documentation/js-api

### PR DESCRIPTION
**Pull request by jgerigmeyer**
- Fixes #218

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
@mirisuzanne I can't figure out why this isn't working -- I just get a Dart Sass error every time I try to use the newer API 😕 

```
    TypeError: J.getInterceptor$ax(...).map$1$1 is not a function

      at Object.map$1$1$ax (node_modules/sass/sass.dart.js:24472:44)
      at listDir_closure0.call$0 (node_modules/sass/sass.dart.js:86789:18)
      at Object._systemErrorToFileSystemException0 (node_modules/sass/sass.dart.js:20737:23)
      at Object.listDir0 (node_modules/sass/sass.dart.js:20732:16)
      at _realCasePath_helper_closure0.call$0 (node_modules/sass/sass.dart.js:84433:34)
      at JsLinkedHashMap.putIfAbsent$2 (node_modules/sass/sass.dart.js:27232:24)
      at _realCasePath_helper0.call$1 (node_modules/sass/sass.dart.js:84422:32)
      at _realCasePath_helper_closure0.call$0 (node_modules/sass/sass.dart.js:84429:35)
      at JsLinkedHashMap.putIfAbsent$2 (node_modules/sass/sass.dart.js:27232:24)
      at _realCasePath_helper0.call$1 (node_modules/sass/sass.dart.js:84422:32)
```

I guess I'll try it out tomorrow on another project?

**Comment by mirisuzanne** [comment]
I don't know much about the new API, but maybe @nex3 can help?

**Comment by jgerigmeyer** [comment]
I'm guessing it has something to do with incorrect paths... I can use `compileString` with simple Sass without errors, but any `@use` statements or trying to use `compile` throws errors.

**Comment by jgerigmeyer** [comment]
@mirisuzanne I'm slowly narrowing it down -- it seems to work properly when run via mocha, but fails via jest.

**Comment by jgerigmeyer** [comment]
So the `jest` tests are still failing for me locally, but I see that they're passing on CI -- maybe something to do with ARM? I'm confused; @mirisuzanne or @jerivas would one or both of you be willing to try running `yarn test` on this project locally?

@mirisuzanne I'd also like your review on the API changes I'm proposing here.

**Comment by jerivas** [comment]
@jgerigmeyer working fine for me on Windows 11 (running inside a `node:lts` container)

<details>
<summary>Output of <code>yarn test</code></summary>

```bash
yarn run v1.22.17
$ run-s build:js test:src test:lib
$ tsc
$ jest
sass/config/_messages.scss:30 DEBUG: ✖ FAILED: [assert-true] This test is meant to fail.
DEBUG: - Output: [bool] false
sass/config/_messages.scss:30 DEBUG: - Expected: [bool] true
: - Module: Throw an error:30 DEBUG
sass/config/_messages.scss:30 DEBUG: - Test: assertionError
WARNING: [assert-true] This test is meant to fail.
    sass/config/_messages.scss 34:7  message()
    sass/data/_details.scss 62:5     fail-details()
    sass/assert/_utils.scss 119:5    result()
    sass/assert/_values.scss 37:3    assert-true()
    - 4:5                            @content
    sass/test/_test.scss 24:3        test()
    - 3:3                            @content
    sass/module/_modules.scss 36:3   test-module()
    - 2:1                            root stylesheet

 PASS  test/main.test.js
 PASS  test/sass.test.js

=============================== Coverage summary ===============================
Statements   : 100% ( 292/292 )
Branches     : 90.64% ( 126/139 )
Functions    : 100% ( 44/44 )
Lines        : 100% ( 289/289 )
================================================================================

Test Suites: 2 passed, 2 total
Tests:       137 passed, 137 total
Snapshots:   0 total
Time:        1.947 s, estimated 2 s
Ran all test suites.
$ USE_BUILT=true mocha


  #fail
    ✔ formats failure message

  #runSass
sass/config/_messages.scss:30 Debug: ✖ FAILED: [assert-true] This test is meant to fail.
sass/config/_messages.scss:30 Debug: - Output: [bool] false
sass/config/_messages.scss:30 Debug: - Expected: [bool] true
sass/config/_messages.scss:30 Debug: - Module: Throw an error
sass/config/_messages.scss:30 Debug: - Test: assertionError
Warning: [assert-true] This test is meant to fail.
    sass/config/_messages.scss 34:7  message()
    sass/data/_details.scss 62:5     fail-details()
    sass/assert/_utils.scss 119:5    result()
    sass/assert/_values.scss 37:3    assert-true()
    - 4:5                            @content
    sass/test/_test.scss 24:3        test()
    - 3:3                            @content
    sass/module/_modules.scss 36:3   test-module()
    - 2:1                            root stylesheet

    ✔ throws AssertionError on failure (38ms)
    ✔ can specify loadPaths (41ms)

  #parse
    ✔ parses a passing non-output test
    ✔ ignores a summary
    ✔ parses a passing non-output test sans description
    ✔ parses a test following a summary
    ✔ parses a nested passing non-output test
    ✔ parses a failing non-output test
    ✔ parses a failing non-output test with no failure details
    ✔ parses a passing output test
    ✔ parses a passing output test with loud comments
    ✔ parses a failing output test
    ✔ respects declaration order in output tests
    ✔ parses tests of comment output
    ✔ ignores unexpected rule types
    ✔ throws error on unexpected rule type instead of end summary
    ✔ accepts a number of context lines to display on error
    ✔ handles a blank comment before module header
    ✔ ignores unexpected rule type instead of test
    ✔ handles a blank comment before test header
    ✔ ignores unexpected rule type instead of assertion
    ✔ handles a blank comment before assertion
    ✔ allows unexpected comment before next module header
    ✔ throws error on unexpected rule type instead of failure detail
    ✔ throws error on unexpected rule type instead of OUTPUT
    ✔ throws error on unexpected comment instead of OUTPUT
    ✔ throws error on unexpected rule type instead of EXPECTED
    ✔ throws error on unexpected comment instead of EXPECTED
    ✔ throws error on unexpected rule type instead of END_ASSERT
    ✔ throws error on unexpected comment instead of END_ASSERT
    #contains
      ✔ parses a passing output test
      ✔ parses a passing output test with loud comments
      ✔ parses a passing output test with curly braces within a content property
      ✔ parses a failing output test
      ✔ parses a failing output test (wrong selector)

  Message
    ✔ Renders messages as CSS comments
    ✔ Renders lists as multiple CSS comments

  True Error [function]
    ✔ Allow errors to return without blocking compilation

  True Error [mixin]
    ✔ Allow errors to output without blocking compilation

  Map Increment
    ✔ Returns a map with the sum-values of two numeric maps

  Join Multiple
    ✔ Combines multiple lists
    ✔ Sets new list-separator

  Context [mixin] & Context-Pop
    ✔ Adds scope and name to context

  Output Context
    ✔ Appends new context
    ✔ Resets context

  Validate Output Context
    ✔ unknown context name
    ✔ duplicate assert
    ✔ duplicate output/expect
    ✔ missing assert
    ✔ wrong length
    ✔ missing context

  Context [function] & Context All
    ✔ Returns current module context
    ✔ Returns current test context
    Context [Nested]
      ✔ Returns the innermost module name
    Context All [Nested]
      ✔ Returns the current stack of module names

  Pass Details
    ✔ Properly output a passing assertion result

  Fail Details
    ✔ Compiles full failure details

  Variable Details
    ✔ Number
    ✔ Color
    ✔ Map
    ✔ Bracketed List

  Edgefail Notes
    ✔ Type mismatch
    ✔ Number Rounding
    ✔ Color rounding
    ✔ List Separators

  Get Result
    ✔ Equal Pass
    ✔ Equal Fail
    ✔ Unequal pass
    ✔ Unequal fail

  Update Results
    ✔ Add one run
    ✔ Add one pass
    ✔ Fail counts are left as-is
    ✔ Output counts are left as-is

  Update Test
    ✔ Updates global test-result
    ✔ Output-to-css overrides pass
    ✔ Pass does not override output-to-css
    ✔ Fail overrides everything
    ✔ Nothing overrides fail

  Results Message
    ✔ Single Line
    ✔ Linebreaks
    ✔ No output tests
    ✔ Single test

  Update Stats Count
    ✔ Assertions counts are updated
    ✔ Modules counts are left as-is
    ✔ Tests counts are left as-is

  Stats Message
    ✔ Single Line
    ✔ Linebreaks

  Setup
    ✔ Updates context based on current assertions

  Is Truthy
    ✔ True is truthy
    ✔ String is truthy
    ✔ List is truthy
    ✔ False is not truthy
    ✔ Null is not truthy
    ✔ Empty string is not truthy
    ✔ Empty list is not truthy

  Assert True
    ✔ Non-false properties return true
    ✔ Supports is-truthy alias

  Assert False
    ✔ Falsiness
    ✔ null
    ✔ Empty string
    ✔ empty list
    ✔ Supports is-falsy alias

  Assert Equal
    ✔ Equality
    ✔ Empty description
    ✔ Adding floats
    ✔ Rounded numbers with $inspect
    ✔ Rounded colors with $inspect
    ✔ Mismatched types with $inspect
    ✔ Supports is-equal alias

  Assert UnEqual
    ✔ Inequality
    ✔ Mismatched types
    ✔ Mismatched units no longer needs $inspect
    ✔ Supports not-equal alias

  Output Expect
    ✔ Compares math output properly
    ✔ Matches output and expected selector patterns

  Output Contains
    ✔ Contains sub-string
    ✔ Contains nested selector block
    ✔ Can be used with nested classes
    ✔ Can be used with nested @media queries
    ✔ Can be used with @at-root
    ✔ Can be used for all selector types
    ✔ Can assert multiple properties within a selector

  Module Title
    ✔ Returns the current module name, prefixed
    Module Title [Nested]
      ✔ Returns a concatenated title of current modules

  Underline
    ✔ Returns a string of dashes, the same length as the input

  Test Module
    ✔ Changes the current module context
    Nested Unquoted Module Name
      ✔ Don’t barf on unquoted names

  Describe
    ✔ Changes the current module context

  Tests
    ✔ Test
    ✔ It [alias]

  Report
    ✔ Output Message
    ✔ Fail on Error
    ✔ Bad results

  Report Message
    ✔ Single Line
    ✔ Linebreaks

  Imported $terminal-output setting
    ✔ Is prefixed as $true-terminal-output


  137 passing (112ms)

Done in 5.20s.
```
</details>

**Comment by jgerigmeyer** [comment]
@jerivas Thanks, that's what I'd expect. Not sure why it's failing for me locally (on any project) -- arm is still my best guess, but it'll be interesting to see whether it works for @mirisuzanne or not. 👀

**Comment by jgerigmeyer** [comment]
[True / Accoutrement Releases](https://trello.com/c/WosNlT5j/67-true-accoutrement-releases)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # True Changelog
 
+## UNRELEASED
```
Comment (mirisuzanne) [inline CHANGELOG.md:3]: Should probably note the new API arguments more clearly here?

**Review by mirisuzanne — COMMENTED** [review]
@jgerigmeyer I am getting the same error as you, also on an M1 mac. The Sass build works fine, tho.

No strong feelings on the API change - though I think the changelog could be a bit more clear about the new shape.

File: CHANGELOG.md (hunk 1 above)
Comment (jgerigmeyer) [reply CHANGELOG.md:3]: Better? https://github.com/oddbird/true/pull/221/commits/3a89271b59e273fa8e4670f7c63afbce108560bb

Comment (mirisuzanne) [reply CHANGELOG.md:3]: lgtm

**Review by mirisuzanne — APPROVED** [review]

**Pull request merged**